access/validator.go

```diff
@@ -346,6 +359,35 @@ func (v *TransactionValidator) checkSignatureFormat(tx *flow.TransactionBody) er
 	return nil
 }
 
+func (v *TransactionValidator) checkSufficientFlowAmountToPayForTransaction(ctx context.Context, tx *flow.TransactionBody) error {
```

Comment:
let's gate running this validator using an option similar to:
https://github.com/onflow/flow-go/blob/b72cbb8405b2cc16b8776b0f79b54f74b0954ff0/access/validator.go#L293